test/jdk/java/foreign/valist/VaListTest.java

```diff
@@ -121,26 +121,26 @@ private static MethodHandle linkVaListCB(String symbol) {
     }
 
     private static final Function<Consumer<VaList.Builder>, VaList> winVaListFactory
-            = actions -> Windowsx64Linker.newVaList(actions, ResourceScope.newConfinedScope());
+            = actions -> Windowsx64Linker.newVaList(actions, MemorySession.openConfined());
     private static final Function<Consumer<VaList.Builder>, VaList> sysvVaListFactory
-            = actions -> SysVx64Linker.newVaList(actions, ResourceScope.newConfinedScope());
+            = actions -> SysVx64Linker.newVaList(actions, MemorySession.openConfined());
     private static final Function<Consumer<VaList.Builder>, VaList> linuxAArch64VaListFactory
-            = actions -> LinuxAArch64Linker.newVaList(actions, ResourceScope.newConfinedScope());
+            = actions -> LinuxAArch64Linker.newVaList(actions, MemorySession.openConfined());
     private static final Function<Consumer<VaList.Builder>, VaList> macAArch64VaListFactory
-            = actions -> MacOsAArch64Linker.newVaList(actions, ResourceScope.newConfinedScope());
+            = actions -> MacOsAArch64Linker.newVaList(actions, MemorySession.openConfined());
     private static final Function<Consumer<VaList.Builder>, VaList> platformVaListFactory
-            = (builder) -> VaList.make(builder, ResourceScope.newConfinedScope());
-
-    private static final BiFunction<Consumer<VaList.Builder>, ResourceScope, VaList> winVaListScopedFactory
-            = (builder, scope) -> Windowsx64Linker.newVaList(builder, scope);
-    private static final BiFunction<Consumer<VaList.Builder>, ResourceScope, VaList> sysvVaListScopedFactory
-            = (builder, scope) -> SysVx64Linker.newVaList(builder, scope);
-    private static final BiFunction<Consumer<VaList.Builder>, ResourceScope, VaList> linuxAArch64VaListScopedFactory
-            = (builder, scope) -> LinuxAArch64Linker.newVaList(builder, scope);
-    private static final BiFunction<Consumer<VaList.Builder>, ResourceScope, VaList> macAArch64VaListScopedFactory
-            = (builder, scope) -> MacOsAArch64Linker.newVaList(builder, scope);
-    private static final BiFunction<Consumer<VaList.Builder>, ResourceScope, VaList> platformVaListScopedFactory
-            = (builder, scope) -> VaList.make(builder, scope);
+            = (builder) -> VaList.make(builder, MemorySession.openConfined());
```

Comment:
Now that the tests no longer close the scope, maybe these scopes should be implicit scopes?